src/query/plan/operator.cpp

```diff
@@ -2372,6 +2375,10 @@ bool EvaluatePatternFilter::EvaluatePatternFilterCursor::Pull(Frame &frame, Exec
   input_cursor_->Reset();
 
   frame[self_.output_symbol_] = TypedValue(input_cursor_->Pull(frame, context), context.evaluation_context.memory);
+  if (context.frame_change_collector &&
```

Comment:
This is a comment that refers to similar diffs below in this file.
Do the .name() here work for lists that are saved using their address?
Is there somewhere we can put the reset that will be called in the same way without copy-pasting this check everywhere?

Reply:
This is only for identifiers. I am not aware of any other way, I think instead of copying I can store it in some kind of util if that works better. Let me know and I can change it if needed.